### ISSUE TITLE
Failing to persist outcomes due to URL vs. DecodedURL confusion

### ISSUE THREAD

**Issue by twm** [twm/yarrharr#619]
An issue much like https://github.com/twisted/treq/issues/282 is biting Yarrharr:

```
[yarrharr.fetch] Failed to persist 5 outcomes
  Traceback (most recent call last):
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
      current.result = callback(current.result, *args, **kw)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/internet/defer.py", line 1475, in gotResult
      _inlineCallbacks(r, g, status)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
      result = result.throwExceptionIntoGenerator(g)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
      return g.throw(self.type, self.value, self.tb)
  --- <exception caught here> ---
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/fetch.py", line 371, in poll
      yield deferToThread(persist_outcomes, outcomes)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/python/threadpool.py", line 250, in inContext
      result = inContext.theWork()
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
      inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/python/context.py", line 122, in callWithContext
      return self.currentContext().callWithContext(ctx, func, *args, **kw)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/twisted/python/context.py", line 85, in callWithContext
      return func(*args,**kw)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/fetch.py", line 630, in persist_outcomes
      outcome.persist(feed)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/fetch.py", line 161, in persist
      self._upsert_article(feed, upsert)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/fetch.py", line 240, in _upsert_article
      created.set_content(upsert.raw_title, upsert.raw_content)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/models.py", line 240, in set_content
      self.content = content = sanitize.sanitize_html(raw_content)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 170, in sanitize_html
      return serializer.render(source)
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/html5lib/serializer.py", line 398, in render
      return "".join(list(self.serialize(treewalker)))
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/html5lib/serializer.py", line 265, in serialize
      for token in treewalker:
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/html5lib/filters/optionaltags.py", line 19, in __iter__
      for previous, token, next in self.slider():
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/html5lib/filters/optionaltags.py", line 10, in slider
      for token in self.source:
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/html5lib/filters/sanitizer.py", line 765, in __iter__
      for token in base.Filter.__iter__(self):
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 456, in _wp_smileys
      for token in source:
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 434, in _video_attrs
      for token in source:
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 416, in _adjust_links
      for token in source:
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 384, in __iter__
      for token in BaseFilter.__iter__(self):
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 352, in __iter__
      ((None, 'href'), self._watch_url(url).to_text()),
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/yarrharr/sanitize.py", line 282, in _watch_url
      query=(('v', video_id),),
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/hyperlink/_url.py", line 838, in __init__
      for k, v in iter_pairs(query))
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/hyperlink/_url.py", line 838, in <genexpr>
      for k, v in iter_pairs(query))
    File "/usr/local/lib/yarrharr/lib/python3.5/site-packages/hyperlink/_url.py", line 475, in _textcheck
      % (''.join(delims), name, value))
  builtins.ValueError: one or more reserved delimiters &# present in query parameter value: 'Nd7exbDzU1c&amp'
```

**Comment by twm** [comment]
See also https://github.com/python-hyper/hyperlink/issues/125

**Comment by twm** [comment]
Fixed in 5dfafff1c8f8b53c6cdf9f23669ef6b3c23b7d65
